code/modules/projectiles/guns/ballistic/pistol.dm

```diff
@@ -85,3 +85,15 @@
 		to_chat(user, "<span class='notice'>..and falls into view. Whew, that was a close one.</span>")
 		user.dropItemToGround(src)
 
+/obj/item/gun/ballistic/automatic/pistol/makeshift
```

Comment:
I'd like it to be designed as a single-shot pistol, similar to a FP-45 Liberator pistol.
Should need to rack the slide, load a new round, rack again

Reply:
This was a bit of a pain, but done!